### PR TITLE
chore: format using make

### PR DESCRIPTION
**Pull request by jonas-jonas**
cc @kevgo 

Just thought, since this repo is really new, I'd add this real quick to have it here. :)

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: Makefile
Comment (kevgo) [inline Makefile:20]: No need to make this target `.PHONY` since there is no `format` file or folder in this codebase.

```suggestion
```

File: Makefile
Comment (kevgo) [inline Makefile:22]: npm has the severe deficiency that it doesn't automatically install newer versions of dependencies when you update the versions in `package.json`. We are therefore using make to do that for us:

```suggestion
format: node_modules
	npm exec -- prettier --write .
	
node_modules: package-lock.json
	npm ci
	touch node_modules
```

Translated into English:
- building the `format` target requires building the `node_modules` target first
- building the `node_modules` target requires building the `package-lock.json` target first
- what was built when is determined by the `updated` timestamp on the filesystem

When somebody updates `package-lock.json`, it will have a newer `updated` time than `node_modules`. Make will therefore rebuild `node_modules` by running `npm ci` and then `touch node_modules` to update the timestamp of that folder.

**Review by kevgo — CHANGES_REQUESTED** [review]
Thanks for implementing this @jonas-jonas! I was indeed unaware of this repo. Please take a look at https://github.com/ory/milestone-action/pull/36/files as a reference for how your PR should look like in the end. Please hit me up on Slack if anything doesn't make sense to you and I'll walk you through it. A few steps to get this to the finish line:

1. Register `prettier` as a development dependency: `npm i --save-dev prettier`
2. Remove the `format` and `format:check` scripts from `package.json`. Nobody should call them directly anymore. We run `make format` instead.
3. Create a new CI job that enforces correct formatting for all future pull requests. You do this by creating a file `.github/workflows/format.yml` with this content:

```yml
name: Format

on:
  pull_request:
  push:

jobs:
  format:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-go@v3
        with:
          go-version: 1.19
      - run: make format
      - name: Indicate formatting issues
        run: git diff HEAD --exit-code --color
```

File: Makefile
Comment (jonas-jonas) [reply Makefile]: Damn, to me `make` always was this obscure "c-era" legacy build tool. But this makes sense and seems to be really well though out. Thanks for the explanation! Appreciate it.

I've updated my PR. Thanks for the pointers.

File: Makefile
Comment (jonas-jonas) [reply Makefile]: Done.

**Comment by jonas-jonas** [comment]
Yes, @Benehiko just created it and I thought I just do it to get the hang of our formatting pipeline. I updated the PR accordingly. Please take another look @kevgo. :)

**Review by kevgo — COMMENTED** [review]
The code is looking so much nicer now! Two more points and then this is done.

File: .github/pull_request_template.md
Comment (jonas-jonas) [reply .github/pull_request_template.md]: Done, to potentially save you some headache in the future: in this repo it's actually `git checkout origin/main ...`... That took me some time to get. ^^

File: CONTRIBUTING.md
Comment (jonas-jonas) [reply CONTRIBUTING.md]: Done.

**Comment by Benehiko** [comment]
We can merge this once @kevgo approves :)

File: .github/pull_request_template.md
Comment (kevgo) [reply .github/pull_request_template.md]: That's only the case if your local `master` branch is not in sync with its remote counterpart at Github (origin/master). To prevent such problems in the future, please install https://www.git-town.com and run `git sync` regularly (like every hour or so).

Comment (jonas-jonas) [reply .github/pull_request_template.md]: no, I always do `git fetch` regularly. :) I meant the `main` instead of `master` part.

**Review by kevgo — APPROVED** [review]
This is perfect now, thanks again for making this! :tada:

File: .github/pull_request_template.md
Comment (kevgo) [reply .github/pull_request_template.md]: Oh right, this repo uses `main`! Also, you are working off a fork.  In this case, please create an `upstream` remote pointing to the original repo:

```
git remote add upstream git@github.com:ory/elements.git
```

Now, when you run `git sync`, it does the right things:

- it fetches from `upstream/main`
- it updates your local branches (main and your feature branch)
- it pushes these updates to your fork an Github

You end up with a setup where all branches in your fork are in sync with the forked repo, and you can therefore use your local branches for all work. No need to ever access remotes in Git commands.

Also, now that your branch is shipped, please run `git prune-branches` to let Git Town remove the now obsolete feature branch your local Git workspace.

**Pull request merged**
